test/mariaex/protocol_test.exs

```diff
@@ -0,0 +1,15 @@
+defmodule Mariaex.ProtocolTest do
+  use ExUnit.Case
+  import ExUnit.CaptureLog
+
+  test "if database goes down, we should handle it" do
+    opts = [database: "information_schema", username: "mariaex_user", password: "mariaex_pass", backoff_type: :rand_exp, sync_connect: true, idle_timeout: 200]
```

Comment:
I think `backoff_type: stop` and then await the `pid` to exit due to disconnect. Will need to trap exits and assert on exit reason received.

Reply:
I prepared this test:

```elixir
  test "if database goes down, we should handle it" do
    opts = [database: "information_schema", username: "mariaex_user", password: "mariaex_pass", backoff_type: :stop, sync_connect: true, idle_timeout: 50]

    Process.flag(:trap_exit, true)

    assert {:ok, pid} = Mariaex.start_link(opts)
    Mariaex.query(pid, "KILL CONNECTION_ID()")

    assert_receive(
      {:EXIT,  ^pid, {:shutdown, %Mariaex.Error{action: "recv", connection_id: ^pid, mariadb: nil, message: nil, reason: :closed, tag: :tcp}}})
  end
```
and it  passes in this branch and also passes on `master`. Which is expected.

I am not able to figure out how to shutdown mysql to stop it emitting pings, because I am addressing the `ping_handle` error response
If I kill the connection (`KILL CONNECTION_ID()`) with `backoff_type: :stop` it will terminate the `Mariaex process, however with `backoff_type: :rand_exp` it should reconnect and not terminate and this happens when handling `error_resp()`

However it's still not clear to me if it would be possible to test this scenario when mariadb server is not pinging back anymore.

See description of this PR for "manual testing"

Thanks!

Reply:
@kpanic there should be no difference between `backoff_type: :stop` and `backoff_type: :rand_exp` for this test/feature. We do not mind about reconnecting, and its easier to wait for the exit because it guarantees what we expected happened. Therefore it seems like we won't be testing this :(.

Reply:
hi @fishcakez yup. I  removed the test commit, would you like to merge it as it is? (without test) Thanks for the support btw :)

Reply:
I didn't mean that we shouldn't test, rather that a difference in `backoff_type` wouldn't change what we wanted to test. I thought you are asserting that we required a different backoff type for tests to pass, and if thats the case the test wasn't testing what it should be testing because backoff type is orthogonal.